great_expectations/execution_engine/sqlalchemy_batch_data.py

```diff
@@ -174,7 +172,7 @@ def __init__(  # noqa: PLR0913
                 )
             )
         elif query:
-            self._selectable = self._generate_selectable_from_query(
+            self._selectable = self._generate_selectable_from_query(  # type: ignore[call-overload] # https://github.com/python/mypy/issues/14764
```

Comment:
The code paths in these conditionals are almost identical.
They could be simplified for v1.